connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/ConnectProtocol.java

```diff
@@ -43,96 +27,9 @@
  * joining the group and distributing assignments, and the format of assignments of connectors and tasks to workers.
  */
 public class ConnectProtocol {
-    public static final String VERSION_KEY_NAME = "version";
-    public static final String URL_KEY_NAME = "url";
-    public static final String CONFIG_OFFSET_KEY_NAME = "config-offset";
-    public static final String CONNECTOR_KEY_NAME = "connector";
-    public static final String LEADER_KEY_NAME = "leader";
-    public static final String LEADER_URL_KEY_NAME = "leader-url";
-    public static final String ERROR_KEY_NAME = "error";
-    public static final String TASKS_KEY_NAME = "tasks";
-    public static final String ASSIGNMENT_KEY_NAME = "assignment";
     public static final int CONNECTOR_TASK = -1;
-
     public static final short CONNECT_PROTOCOL_V0 = 0;
```

Comment:
The `CONNECT_PROTOCOL_V0` is only used in `ConnectProtocolCompatibility.fromProtocolVersion()`, I think we shouldn't depend on the version to compute `ConnectProtocolCompatibility` because every time we bump the version we should change the method, maybe we can add a field in the ExtendedWorkerMetadata.json to represent the corresponding `ConnectProtocolCompatibility`.

Reply:
I feel the version is bump only if we add new protocol (see https://github.com/apache/kafka/blob/trunk/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java#L190). 

For another, your comment inspires me that the constant field can be moved to ```ConnectProtocolCompatibility``` as it is used by ```ConnectProtocolCompatibility``` only